src/onboarding/registration/ProtectWallet.tsx

```diff
@@ -0,0 +1,241 @@
+import { NativeStackScreenProps } from '@react-navigation/native-stack'
+import React, { useLayoutEffect, useState } from 'react'
+import { useAsync } from 'react-async-hook'
+import { useTranslation } from 'react-i18next'
+import { ActivityIndicator, View, ScrollView, StyleSheet, Text } from 'react-native'
+import { SafeAreaView } from 'react-native-safe-area-context'
+import { useSelector } from 'react-redux'
+import { OnboardingEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import Button, { BtnSizes, BtnTypes } from 'src/components/Button'
+import { HeaderTitleWithSubtitle, nuxNavigationOptionsOnboarding } from 'src/navigator/Headers'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { StackParamList } from 'src/navigator/types'
+import GuideKeyIcon from 'src/icons/GuideKeyIcon'
+import { getOnboardingStepValues, onboardingPropsSelector } from 'src/onboarding/steps'
+import { default as useTypedSelector } from 'src/redux/useSelector'
+import colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import OnboardingCard from 'src/components/OnboardingCard'
+import { twelveWordMnemonicEnabledSelector, walletAddressSelector } from 'src/web3/selectors'
+import seedrandom from 'seedrandom'
+import BottomSheet from 'src/components/BottomSheet'
+import { getWalletAsync } from 'src/web3/contracts'
+import { getPassword } from 'src/pincode/authentication'
+import { UnlockableWallet } from '@celo/wallet-base'
+import { UNLOCK_DURATION } from 'src/web3/consts'
+import variables from 'src/styles/variables'
+
+type Props = NativeStackScreenProps<StackParamList, Screens.ProtectWallet>
+
+function getCloudBackupIndex(seed: string | null) {
+  if (!seed) {
+    return 1
+  } else {
+    return seedrandom(seed)() > 0.5 ? 0 : 1
+  }
+}
+
+function ProtectWallet({ navigation }: Props) {
+  const twelveWordMnemonicEnabled = useSelector(twelveWordMnemonicEnabledSelector)
+  const mnemonicLength = twelveWordMnemonicEnabled ? '12' : '24'
+  const onboardingProps = useTypedSelector(onboardingPropsSelector)
+  const { step, totalSteps } = getOnboardingStepValues(Screens.ProtectWallet, onboardingProps)
+  const address = useSelector(walletAddressSelector)
+  const [showBottomSheet, setShowBottomSheet] = useState(false)
+  const [shouldNavigate, setShouldNavigate] = useState(false)
+  const { t } = useTranslation()
+
+  const cloudBackupIndex = getCloudBackupIndex(address)
+
+  // We need to ensure that the wallet is unlocked before navigating to the next screen,
```

Comment:
An alternative here would be to pass the mnemonic as a prop to the next screen; during testing, this was a bit of an issue if the wallet was locked when navigating to this page, as calling `useAccountKey` prompts the user for their PIN if the wallet is locked. We can't call `useAccountKey` "on demand" when the user presses the button to go to the next screen, since React will yell at you if a different set of hooks is called on subsequent renders.

In practice, however, it's probably impossible for the user's account to be locked when arriving at this screen, so we could in theory just _assume_ that the user's account is unlocked here, call `useAccountKey` immediately, and pass the result as a prop to the next page. _If_ it so happens that the user's account is locked for whatever reason however, this would cause problems.

Reply:
Would the users account be locked if they completed entering their pin, arrived at this screen, then set down their phone for an hour?

Reply:
Also, what about using the same approach that the settings drawer uses to gate whether or not a user can see their recovery phrase with `ensurePincode`? https://github.com/valora-inc/wallet/blob/main/src/navigator/DrawerNavigator.tsx#L119-L128

Reply:
wow, i wish i had known about this sooner!! `ensurePincode` is exactly what we want here, and way cleaner. will replace with that. and to answer your question -- yeah, that's definitely a possibility.